src/pivot-table/data/extract-left.ts

```diff
@@ -25,10 +26,14 @@ const extractLeftGrid = (
       grid[colIdx] = {};
     }
 
-    nodes.forEach((node, currRowIdx) => {
-      rowIdx += currRowIdx === 0 ? 0 : 1;
-      const y = qArea.qTop + rowIdx - node.qUp; // Start position + current page position - previous tail size,
-      const cell = createCell(node, parent, root, colIdx, y, isSnapshot);
+    nodes.forEach((node, idx) => {
+      rowIdx += idx === 0 ? 0 : 1;
+      // consider items that might be skipped based on current page
+      const startPosition = qArea.qTop - pageInfo.currentPage * pageInfo.rowsPerPage;
+      // Start position + current page position - previous tail size,
+      const y = Math.max(0, startPosition + rowIdx - node.qUp);
+      const dataY = qArea.qTop + rowIdx - node.qUp;
```

Comment:
I do think you should call it `y` and `pageY`, so that both `x`and `y` refers to the "data" position. No need to fix in this PR.

Reply:
will fix it soon in another PR